### PR TITLE
manually deleting temp dirs for iterations - skipping Specimin except…

### PR DESCRIPTION
**Pull request by jonathan-m-phillips**
…ion if in dryrun - made the model a global state in Ashe.java so the entire application can utilize it without passing it around.

### CODE REVIEW

File: src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java [hunk 1]
```diff
@@ -40,14 +45,14 @@ private SpeciminTool() {
...
-     * @param targetMethod Method to be targeted by the tool.
-     * @return The directory path where the minimized file is saved.
-     * @throws IOException          If there's an error executing the command or writing the minimized file.
-     * @throws InterruptedException If the process execution is interrupted.
+     * @param root         the root directory for the tool
+     * @param targetFile   file to be targeted by the tool
+     * @param targetMethod method to be targeted by the tool
+     * @return the directory path where the minimized file is saved
+     * @throws IOException          if there's an error executing the command or writing the minimized file
+     * @throws InterruptedException if the process execution is interrupted
```
Comment (mernst) [inline src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java:55]: These are good changes, thanks.
It is better to make a separate pull request for each separate concern.  So, these would be better in a separate pull request.

File: src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java [hunk 2]
```diff
@@ -69,13 +74,43 @@ public static String runSpeciminTool(String root, String targetFile, String targ
     }
 
     /**
-     * Creates a temporary directory for storing output from the Specimin tool.
+     * Deletes all files and subdirectories in the Specimin temporary directory. This method is specifically
```
Comment (mernst) [inline src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java:77]: "in the Specimin temporary directory":  Is this true?  It seems to work on any given directory, and so the documentation and method name need to be updated.

File: src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java [hunk 3]
```diff
@@ -69,13 +74,43 @@ public static String runSpeciminTool(String root, String targetFile, String targ
...
+     * important when automating {@link Ashe} with {@link AsheAutomation} and {@link RepositoryAutomationEngine}.
+     * If the temporary directory is not deleted, many directories will be created and require a lot of
+     * disk space.
      *
-     * @return Path of the created temporary directory.
+     * @param tempDirPath the {@link Path} to the directory that needs to be deleted
+     * @throws IOException if an I/O error occurs during file operations
+     */
+    public static void deleteSpeciminTempDir(Path tempDirPath) throws IOException {
```
Comment (mernst) [inline src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java:85]: It really stinks that Java doesn't provide a method for deleting a directory.
Rather than re-inventing the wheel, how about using `deleteDir()` from https://plumelib.org/plume-util/api/org/plumelib/util/FilesPlume.html or `FileUtils.deleteDirectory()` from commons-io?

File: src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java [hunk 4]
```diff
@@ -126,7 +161,7 @@ private static List<String> prepareCommands(String speciminPath, String argsWith
     /**
      * Logs the commands being executed.
      *
-     * @param commands List of commands to be logged.
+     * @param commands {@link List} of commands to be logged
```
Comment (mernst) [inline src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java:164]: This `{@link ...}` is not needed, because in the Javadoc, the return type will be linked.  The same comment goes for anywhere else that there is a `{@link ...}` for a type that is in a signature.

File: src/main/java/edu/njit/jerse/ashe/Ashe.java [hunk 5]
```diff
@@ -92,40 +99,46 @@ public static void run(String root, String targetFile, String targetMethod, Stri
...
+            boolean errorsReplacedInTargetFile = corrector.fixTargetFileErrorsWithModel(sourceFilePath, targetMethod, model);
+
+            if (!errorsReplacedInTargetFile) {
+                if (corrector.checkedFileError(sourceFilePath).isEmpty()) {
+                    LOGGER.info("No errors found in the file, no replacements needed.");
+                    LOGGER.info("Exiting...");
+                    return;
+                }
+                LOGGER.error("Errors were found but not replaced with {} response.", model);
+                throw new RuntimeException("Errors were not replaced with " + model + " response.");
```
Comment (mernst) [inline src/main/java/edu/njit/jerse/ashe/Ashe.java:121]: I suggest creating a variable to hold the string, then re-using the variable.  That makes it perfectly clear that the two strings are the same, and it prevents them from getting out of sync.

**Review by mernst — CHANGES_REQUESTED** [review]

File: src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java (hunk 3 above)
Comment (jonathan-m-phillips) [reply src/main/java/edu/njit/jerse/ashe/services/SpeciminTool.java:85]: Just using deleteDir() within Ashe#run is much better than creating this method. The change has been made and SpeciminTool.java no longer holds this method. Additionally, we actually do not need createTempDirectory() either and just added noise. That too was deleted.

File: src/main/java/edu/njit/jerse/ashe/Ashe.java (hunk 5 above)
Comment (jonathan-m-phillips) [reply src/main/java/edu/njit/jerse/ashe/Ashe.java:121]: I will make these changes throughout the project in a separate PR specific to this requested change.

**Review by mernst — APPROVED** [review]
